client/client-benchmark-noop-api-plugin/src/main/java/org/elasticsearch/plugin/noop/action/bulk/RestNoopBulkAction.java

```diff
@@ -71,7 +71,7 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         if (waitForActiveShards != null) {
             bulkRequest.waitForActiveShards(ActiveShardCount.parseString(waitForActiveShards));
         }
-        bulkRequest.timeout(request.paramAsTime("timeout", BulkShardRequest.DEFAULT_TIMEOUT));
+        bulkRequest.timeout(request.paramAsTime("shard_timeout", BulkShardRequest.DEFAULT_TIMEOUT));
```

Comment:
We'll need to support the old name and log a deprecation warning if it is used for a while. Usually it is easier to write the "deprecate" PR first and get that is and *then* merge the removal PR on top of that. The "deprecate" PR can be merged to both master and then 5.x, but the "remove" PR really can only go in master.